### PR TITLE
ci: Add a workflow to publish crate to crate.io

### LINKED ISSUE #76: Publish crate to crate.io
Publish crate to crate.io

Need source dev account to generate the API token.

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
Resolves #76 

## Description
- Adds workflow to publish crate according to our needs
- Manual trigger only, on as needed basis

### CODE REVIEW

File: .github/workflows/publish-crate.yml
Comment (AndrewSisley) [inline .github/workflows/publish-crate.yml:14]: Note: `master` doesn't exist in this repo (`main` is just the new default branch for more recent git versions).

thought: I'm really not sure we want to publish automatically to main.  As well as whether we want to do this, if the version of the package we are publishing has not changed in the new commit, the `cargo publish` command will fail, as the version already exists in the package repository (crates.io).

File: .github/workflows/publish-crate.yml
Comment (AndrewSisley) [inline .github/workflows/publish-crate.yml:45]: question:  This looks very insecure, is it?

Comment (shahzadlone) [reply .github/workflows/publish-crate.yml:45]: Is pretty standard practice: https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#example-using-bash

Here is how the env var will show up in logs:
```
  env:
    CRATES_TOKEN: ***
```

However I noticed with cargo.io another practice can be to just set the environment var `CARGO_REGISTRY_TOKEN` with the secret which can pick it up. I can try that if you prefer it?

Comment (AndrewSisley) [reply .github/workflows/publish-crate.yml:45]: Ah okay nice, and thanks for the explanation :)  I guess github must do some magic here then - no need to change, happy as is :)

File: .github/workflows/publish-crate.yml
Comment (shahzadlone) [reply .github/workflows/publish-crate.yml:14]: There is an action that handles some additional checks, but was hesitant to use it as it's not that popular. Might be worth a look: https://github.com/katyo/publish-crates

In that case can leave this to be a dispatch trigger only for now (manually).

**Comment by shahzadlone** [comment]
Here is a successful run that doesn't upload as for the current tag the crate is already pushed.

All else is done, the trigger is now manual.

@AndrewSisley you can try it out manually through the actions tab (it should be fine as long as the workflow file in on the default branch. i.e. main branch in this case).

File: .github/workflows/publish-crate.yml
Comment (shahzadlone) [reply .github/workflows/publish-crate.yml]: I did end up changing it, it does seem cleaner.

**Comment by shahzadlone** [comment]
To manually trigger, after this PR is merged you will be shown a trigger button at:
https://github.com/lens-vm/lens/actions/workflows/publish-crate.yml

**Comment by AndrewSisley** [comment]
> To manually trigger, after this PR is merged you will be shown a trigger button at: https://github.com/lens-vm/lens/actions/workflows/publish-crate.yml

Cheers, an on-demand workflow is much better than passing round creds or pushing with our users.  You should probably remove mine and your users as owners of the crate now too.

**Review by AndrewSisley — APPROVED** [review]
LGTM - thanks Shahzad :)

**Pull request merged**
